### ISSUE TITLE
celery long running nested chain in chord on multi node env - chord callback not executed

### ISSUE THREAD

**Issue by liorchen** [celery/celery#5027]
## Environment
```
➜  src celery -A service report
/Users/liorchen/.pyenv/versions/3.6.3/envs/gndn.celery/lib/python3.6/site-packages/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.
  """)

software -> celery:4.2.1 (windowlicker) kombu:4.2.1 py:3.6.3
            billiard:3.5.0.4 py-amqp:2.3.2
platform -> system:Darwin arch:64bit imp:CPython
loader   -> celery.loaders.app.AppLoader
settings -> transport:pyamqp results:redis://localhost:6379/0

Exchange: <class 'kombu.entity.Exchange'>
Queue: <class 'kombu.entity.Queue'>
accept_content: ['msgpack']
broker_url: 'amqp://guest:********@127.0.0.1:5672//'
enable_utc: True
imports: 
    ('service.unloader', 'service.scoring', 'service.opencpu')
os: <module 'os' from '/Users/liorchen/.pyenv/versions/3.6.3/lib/python3.6/os.py'>
result_backend: 'redis://localhost:6379/0'
result_persistent: False
result_serializer: 'msgpack'
task_queues: 
    (<unbound Queue unloader.cpubound -> <unbound Exchange gndn(direct)> -> >,
 <unbound Queue unloader.gevent -> <unbound Exchange gndn(direct)> -> >,
 <unbound Queue opencpu.gevent -> <unbound Exchange gndn(direct)> -> >,
 <unbound Queue scoring.gevent -> <unbound Exchange gndn(direct)> -> >)
task_routes: {
    'service.opencpu.*': {'exchange': 'gndn', 'queue': 'opencpu.gevent'},
    'service.scoring.*': {'exchange': 'gndn', 'queue': 'scoring.gevent'},
    'service.unloader.do_refragementation_of_unloaded_csv': {   'exchange': 'gndn',
                                                                'queue': 'unloader.gevent'},
    'service.unloader.do_unload': {   'exchange': 'gndn',
                                      'queue': 'unloader.cpubound'},
    'service.unloader.get_unload_query': {   'exchange': 'gndn',
                                             'queue': 'unloader.cpubound'},
    'service.unloader.unload_from_redshift': {   'exchange': 'gndn',
                                                 'queue': 'unloader.cpubound'}}
task_serializer: 'msgpack'
```

## Steps to reproduce
1. implement a task which sets up a chain of multiple tasks (in my situation it was 3 with the the first two assigned to seprated queue), one of these tasks should take a long time (3-5 minutes).
1. in addtion our task would setup a signature of another task that should run in parrallel to the chain.
1. then declare a chord(ourchain, oursingletask)(callback)
1. return the AsyncResult.id of the chord
1. start 2 nodes when each of them is responsible for different queue.
1. call the root task, chord callback never executes.

example from my code:

```
unload_flow = chain(get_unload_query.s(*gen_query_params),
                        unload_from_redshift.s(**unload_params),
                        do_refragementation_of_unloaded_csv.s(100000))
load_models_signature = load_models_multiservers \
        .s(servers, models_meta, models_bucket, load_models_extra)
c = chord((unload_flow,
                 load_models_signature
               ))(classify_results.s(models_meta, output_headers, classify_extra, s3_target_params))
return c.id
```

## Expected behavior
chord callback should always execute even if distributed tasks takes lots of time

## Actual behavior
chord callback not executed. This thing works only on a single celery node/shorter execution time.

Lior Chen

**Comment by auvipy** [comment]
status of this with celery 4.3 or master?

**Comment by liorchen** [comment]
@auvipy after long investigations, I found that the core problem goes like that:
1. you run long running operations in canvas (I've tried both chord/chain).
2. there are multiple flows running in parallel.
3. the broker is rabbitmq

The result:
some of the tasks never executes without any feedback. i.e a chain of tasks only executed partially, and in chord, the callback never executes.

We ended up choosing redis broker, but it's really disappointing..
